lte/gateway/python/magma/subscriberdb/protocols/m5g_auth_servicer.py

```diff
@@ -179,11 +179,11 @@ def M5GDecryptMsinSUCIRegistration(self, request, context):
                 profile,
             )
 
-            if len(request.ue_pubkey) != 32:
+            if len(request.ue_pubkey) != 32 and len(request.ue_pubkey) != 33:
```

Comment:
Should we check here that the key length fits the profile (32 for A and 33 for B if I understand it correctly)?

Reply:
Yes it is needed because unprotect function needs key length 32 bytes in case of Profile A and 33 bytes in case of Profile B.

Reply:
But with the current implementation we don't check the length depending on the profile. So we would allow 33 bytes for profile A.

Reply:
Added check for public key length depending on the profile.